File: 프로그래머스/2/64065. 튜플/튜플.java
import java.util.*;
class Solution {
    public int[] solution(String s) {
        
        s = s.replace("{","");
        s = s.replace("}","");
        HashMap <Integer,Integer> map = new HashMap<>();
        String[] str = s.split(",");
        for(int i = 0; i < str.length; i++) {
            int n = Integer.parseInt(str[i]);
            map.put(n,map.getOrDefault(n,0)+1);
        }
        int[] answer = new int[map.size()];

        for(int key : map.keySet()) {
            answer[map.size() - map.get(key)] = key;
        }
        
        
        
        
        return answer;
    }
}

File: 프로그래머스/2/64065. 튜플/튜플.js
function solution(s) {
    var answer = [];
    s = s.replaceAll('{','');
    s = s.replaceAll('}','');
    s = s.split(',').map((v)=>Number(v));
    const map = {};
    for (let i = 0; i < s.length; i++) {
        const num = s[i];
        map[num] = (map[num] || 0) + 1;
    }
    answer = Object.entries(map).sort((a,b) => b[1] - a[1]);
    answer = answer.map((v) => Number(v[0]));
    return answer;
}